src/api.js

```diff
@@ -478,14 +479,44 @@ let ensureTaskGroup = async (ctx, taskId, taskDef, res) => {
   await ctx.TaskGroupMember.create({
     taskGroupId,
     taskId,
-    expires: new Date(taskDef.expires),
+    expires,
   }).catch(err => {
     // If the entity already exists, then we're happy no need to crash
     if (!err || err.code !== 'EntityAlreadyExists') {
       throw err;
     }
   });
 
+  // Now we also add the task to the group size counters as well
+  await ctx.TaskGroupActiveSet.create({
+    taskGroupId,
+    taskId,
+    expires,
+  }).catch(async (err) => {
+    // If the entity already exists, then we're happy no need to crash
+    if (!err || err.code !== 'EntityAlreadyExists') {
+      throw err;
+    }
+
+    let active = await ctx.TaskGroupActiveSet.load({taskId, taskGroupId});
+
+    if (!_.isEqual(new Date(active.expires), expires)) {
+      return res.reportError('RequestConflict', [
+        'TaskGroupActiveSet for taskId:{{taskId}} taskGroupId:{{taskGroupId}}',
```

Comment:
Don't mention `TaskGroupActiveSet`, just say `taskId` already in use... `TaskGroupActiveSet` is an internal thing...